### PR TITLE
sinkmanager(ticdc): add table sink wrapper

### PR DESCRIPTION
**Pull request by hi-rustin**
### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tiflow/issues/5928

### What is changed and how it works?

- add table sink wrapper
- add table progress heap

This PR introduces the table sink wrapper, and it is used to emit data to the table sink. It is difficult to reuse the sink node so we have to add a new wrapper.

And we introduce a progress heap to track the fetch progress. It is a thread-safe heap.

#### Tests 

 - Unit test

##### Will it cause performance regression or break compatibility?
No
##### Do you need to update user documentation, design documentation or monitoring documentation?
No
### Release note 

```release-note
None
```

### CODE REVIEW

**Review by nongfushanquan — COMMENTED** [review]
Please add some comments to describe the purpose of this pr

**Comment by hi-rustin** [comment]
> Please add some comments to describe the purpose of this pr

Added. Thanks for your reminder.

File: cdc/processor/sinkmanager/table_sink_wrapper.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	log.Info("sink is closed",
+		zap.Int64("tableID", t.tableID),
+		zap.String("namespace", t.changefeed.Namespace),
+		zap.String("changefeed", t.changefeed.ID))
+	return cerror.ErrTableProcessorStoppedSafely.GenWithStackByArgs()
+}
+
+// convertRowChangedEvents uses to convert RowChangedEvents to TableSinkRowChangedEvents.
+// It will deal with the old value compatibility.
+func convertRowChangedEvents(changefeed model.ChangeFeedID, tableID model.TableID, enableOldValue bool, events ...*model.PolymorphicEvent) ([]*model.RowChangedEvent, error) {
```
Comment (nongfushanquan) [inline cdc/processor/sinkmanager/table_sink_wrapper.go:109]: is it reasonable to put this method in this struct?

Comment (hi-rustin) [reply cdc/processor/sinkmanager/table_sink_wrapper.go:109]: No, it will also be used to redo tasks. It is more like a util.

**Review by hicqu — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Pull request merged**